### PR TITLE
DAT-2975 add width & height params to infobox image

### PR DESCRIPTION
**Pull request by dianafa**
https://wikia-inc.atlassian.net/browse/DAT-2975

### CODE REVIEW

File: extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php [hunk 1]
```diff
@@ -95,8 +95,14 @@ private function renderItem( $type, array $data ) {
 		//TODO: with validated the performance of render Service and in the next phase we want to refactor it (make
 		// it modular) While doing this we also need to move this logic to appropriate image render class
 		if ( $type === 'image' ) {
-			$data[ 'thumbnail' ] = $this->getThumbnailUrl( $data['name'] );
 			$data[ 'key' ] = urlencode( $data[ 'key' ] );
+			$thumbnail = $this->getThumbnail( $data['name'] );
+
+			if ( $thumbnail ) {
```
Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php]: What if not? :)

File: extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceTest.php [hunk 2]
```diff
@@ -1,14 +1,32 @@
...
+class PortableInfoboxRenderServiceTest extends WikiaBaseTest {
 	private $infoboxRenderService;
 
 	protected function setUp() {
 		$this->setupFile = dirname( __FILE__ ) . '/../PortableInfobox.setup.php';
 		parent::setUp();
 
-		$mock = $this->getMock( 'PortableInfoboxRenderService', [ 'getThumbnailUrl' ] );
-		$mock->expects( $this->any() )->method( 'getThumbnailUrl' )->will( $this->returnValue( 'http://image.jpg' ) );
+		$mockThumbnailImage = $this->getMockBuilder( 'ThumbnailImage' )
```
Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceTest.php]: Let's support also the case when there is no thumbnail (i.e. MediaTransformError)

Comment (dianafa) [inline extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceTest.php]: We do support it here: https://github.com/Wikia/app/pull/7805/files#diff-38f4ac017c46ec2a05d85a08356fab19R131

File: extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php (hunk 1 above)
Comment (dianafa) [inline extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php]: now is consistent with previous `getThumbnailUrl` behaviour.

Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php]: :+1:

File: extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceTest.php (hunk 2 above)
Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceTest.php]: Yes, I was referring to the test (that's why I changed support => cover in the comment). Thanks!

Comment (dianafa) [inline extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceTest.php]: In this case the getThumbnail will return false (simply logic is used there, didn't want to dig deeper in the thumbnailer error throwing functionality as it is not the subject of this test)

Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceTest.php]: Ok, I see you added a test case for that. That covers what I had in mind, thanks! :)

File: extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceTest.php [hunk 3]
```diff
@@ -48,10 +72,10 @@ public function testRenderInfobox( $input, $expectedOutput, $description ) {
 	 * @param $input
 	 * @param $expectedOutput
 	 * @param $description
-	 * @dataProvider testRenderInfoboxDataProvider
+	 * @dataProvider testRenderMobileInfoboxDataProvider
 	 */
```
Comment (dianafa) [inline extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceTest.php]: this test was always run with wrong data till now :D :dancers:

File: extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php [hunk 4]
```diff
@@ -86,17 +86,26 @@ private function renderGroup( $groupData ) {
 
 	/**
 	 * renders part of infobox
+	 * If image element has invalid thumbnail, doesn't render this element at all.
 	 *
 	 * @param string $type
 	 * @param array $data
-	 * @return string - HTML
+	 * @return bool|false|string - HTML
```
Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php]: in this case the convention would be bool|string

Comment (dianafa) [inline extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php]: autogenerated :P

File: extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceTest.php [hunk 5]
```diff
@@ -25,8 +60,7 @@ private function setWikiaMobileSkin($bool) {
 	 * @dataProvider testRenderInfoboxDataProvider
 	 */
 	public function testRenderInfobox( $input, $expectedOutput, $description ) {
-		$this->setWikiaMobileSkin(false);
-
+		$this->infoboxRenderService = $this->getInfoboxRenderServiceMock( false, $input );
```
Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceTest.php]: It'll be safer to stop using $this-> here and instantiate $infoboxRenderService in the scope of each test

File: extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceTest.php [hunk 6]
```diff
@@ -1,21 +1,56 @@
...
 		$this->setupFile = dirname( __FILE__ ) . '/../PortableInfobox.setup.php';
 		parent::setUp();
+	}
+
+	/**
+	 * @param $isWikiaMobile
+	 * @param $input to check presence of 'invalidImage' field
+	 * @return PHPUnit_Framework_MockObject_MockObject
+     */
+	private function getInfoboxRenderServiceMock( $isWikiaMobile, $input )
```
Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceTest.php]: Per offline discussion, let's use a consistent type of configuring the mock - two bools, an associative config array, or an interface similar to one in https://github.com/Wikia/app/blob/DAT-2983/extensions/wikia/ImageServing/tests/ImageServingDriverMainNSTest.php#L49-L55

File: extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceTest.php [hunk 7]
```diff
@@ -25,8 +60,7 @@ private function setWikiaMobileSkin($bool) {
 	 * @dataProvider testRenderInfoboxDataProvider
 	 */
 	public function testRenderInfobox( $input, $expectedOutput, $description ) {
-		$this->setWikiaMobileSkin(false);
-
+		$this->infoboxRenderService = $this->getInfoboxRenderServiceMock( false, $input );
 		$actualOutput = $this->infoboxRenderService->renderInfobox( $input );
 
 		$actualDOM = new DOMDocument('1.0');
```
Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceTest.php]: I know you didn't change this line, but I see this block duplicated in the test - any chances to extract the normalizeHTML() function?

**Comment by SebastianMarzjan** [comment]
Two small naming suggestions.
LGTM and thanks for cleaning up so much! :)

**Comment by dianafa** [comment]
thank you for patience! :lollipop:

**Comment by SebastianMarzjan** [comment]
Looks great to axolotl!
:+1:

Looks great to axolotl

**Pull request merged**